lib/src/entrypoint.dart

```diff
@@ -916,6 +931,113 @@ See https://dart.dev/go/sdk-constraint
+        break;
+      }
+      final cause = externalIncompatibility.cause;
+      if (cause is SdkCause) {
+        final sdkName = cause.sdk.identifier;
+        if (!(sdkName == 'dart' ||
+            (sdkName == 'flutter' && runningFromFlutter))) {
+          // Only make sdk upgrade suggestions for Flutter and Dart.
+          continue;
+        }
+
+        final constraint = cause.constraint;
+        if (constraint == null) continue;
+
+        /// Find the most relevant Flutter release fullfilling the constraint.
+        final bestRelease =
+            await inferBestFlutterRelease({cause.sdk.identifier: constraint});
+        if (bestRelease == null) continue;
+        try {
+          await resolveVersions(type, cache, root,
+              lockFile: lockFile,
+              unlock: unlock,
+              sdkOverrides: {
+                'dart': bestRelease.dartVersion,
+                'flutter': bestRelease.flutterVersion
+              });
+
+          if (runningFromFlutter) {
+            suggest(
+                '* Try using the Flutter SDK version: ${bestRelease.flutterVersion}. ');
+          } else {
+            // Here we assume that any Dart version included in a Flutter
+            // release can also be found as a released Dart SDK.
+            suggest(
+                '* Try using the Dart SDK version: ${bestRelease.dartVersion}. See https://dart.dev/get-dart.');
+          }
+        } on SolveFailure {
+          // Using a newer sdk didn't work.
+          // Nothing to report.
+        }
+      } else {
+        for (final term in externalIncompatibility.terms) {
+          final name = term.package.name;
+          if (!visited.add(name)) {
+            continue;
+          }
+          final originalConstraint =
+              (root.dependencies[name] ?? root.devDependencies[name])
+                  ?.constraint;
+          if (originalConstraint != null) {
+            final relaxedPubspec = stripVersionBounds(root.pubspec,
+                stripOnly: [name], stripLowerBound: true);
+
+            try {
+              final result = await resolveVersions(
+                  type, cache, Package.inMemory(relaxedPubspec),
+                  lockFile: lockFile, unlock: unlock);
+              final resolvedVersion =
+                  result.packages.firstWhere((p) => p.name == name).version;
+
+              final newConstraint =
+                  VersionConstraint.compatibleWith(resolvedVersion);
+
+              String updateDirection = 'updating';
+              if (originalConstraint is VersionRange) {
+                final min = originalConstraint.min;
+                if (min != null) {
+                  updateDirection =
+                      resolvedVersion < min ? 'downgrading' : 'upgrading';
+                }
+              }
+
+              suggest(
+                  '* Try $updateDirection your constraint on $name: `$topLevelProgram pub add $name:$newConstraint`');
```

Comment:
Maybe we can make it easier to copy/paste
```suggestion
                  '* Try $updateDirection your constraint on $name: $topLevelProgram pub add $name:$newConstraint');
```

Could the message be: `dart pub upgrade --major-version $name`? I'm guessing it's not better, because it does say what version you'll be getter in the message / command, so `dart pub add` is probably better.

Reply:
Done